src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -381,32 +381,14 @@ private static GpioDriver GetBestDriverForBoard()
         /// <returns>A driver that works with the board the program is executing on.</returns>
         private static GpioDriver GetBestDriverForBoardOnLinux()
         {
-            string[] cpuInfoLines = File.ReadAllLines(CpuInfoPath);
-            Regex regex = new Regex(@"Hardware\s*:\s*(.*)");
-            foreach (string cpuInfoLine in cpuInfoLines)
+            try
             {
-                Match match = regex.Match(cpuInfoLine);
-                if (match.Success)
-                {
-                    if (match.Groups.Count > 1)
-                    {
-                        if (match.Groups[1].Value == RaspberryPiHardware)
-                        {
-                            return new RaspberryPi3Driver();
-                        }
-
-                        // Commenting out as HummingBoard driver is not implemented yet, will be added back after implementation
-                        // https://github.com/dotnet/iot/issues/76
-                        // if (match.Groups[1].Value == HummingBoardHardware)
-                        // {
-                        //     return new HummingBoardDriver();
-                        // }
-                        return UnixDriver.Create();
-                    }
-                }
+                return new RaspberryPi3Driver();
```

Comment:
can you create some helper which you could use here and in the constructor to avoid catching here?

Reply:
Yea, done. Is a bit messy as well, since the ctor of the `RaspberryPi3Driver` is public and should work even when called directly (in which case an exception is ok).

Reply:
PS: I'm more and more in favor of getting the board identification stuff correctly done and dropping outside access to the drivers. We're doing more and more workarounds to keep the API static.

Reply:
I agree with you @pgrawehr. I'm sure very soon a discussion will pop up to address the Gpio API